db/logs.go

```diff
@@ -63,3 +63,14 @@ func (d *LogsDB) Logs(ctx context.Context, dealID uuid.UUID) ([]DealLog, error)
 
 	return dealLogs, nil
 }
+
+func (d *LogsDB) CleanupLogs(ctx context.Context, duration int) error {
```

Comment:
nit: suggest renaming the duration parameter to make clear the unit is days:
```suggestion
func (d *LogsDB) CleanupLogs(ctx context.Context, daysOld int) error {
```